kws/model/mdtc.py

```diff
@@ -0,0 +1,258 @@
+            in_channels=in_channels,
+            out_channels=res_channels,
+            kernel_size=kernel_size,
+            dilation=dilation,
+        )
+        self.bn1 = nn.BatchNorm1d(res_channels)
+        self.relu1 = nn.ReLU()
+
+        self.conv2 = nn.Conv1d(in_channels=res_channels,
+                               out_channels=res_channels,
+                               kernel_size=1)
+        self.bn2 = nn.BatchNorm1d(res_channels)
+        self.relu2 = nn.ReLU()
+
+    def forward(self, inputs: torch.Tensor):
+        outputs = self.relu1(self.bn1(self.conv1(inputs)))
+        outputs = self.bn2(self.conv2(outputs))
+        if self.causal:
+            inputs = inputs[:, :, self.receptive_fields:]
+        else:
+            inputs = inputs[:, :, self.
+                            half_receptive_fields:-self.half_receptive_fields]
+        if self.in_channels == self.res_channels:
+            res_out = self.relu2(outputs + inputs)
+        else:
+            res_out = self.relu2(outputs)
+        return res_out
+
+
+class TCNStack(nn.Module):
+    def __init__(
+        self,
+        in_channels: int,
+        stack_num: int,
+        stack_size: int,
+        res_channels: int,
+        kernel_size: int,
+        causal: bool,
+    ):
+        super(TCNStack, self).__init__()
+        self.in_channels = in_channels
+        self.stack_num = stack_num
+        self.stack_size = stack_size
+        self.res_channels = res_channels
+        self.kernel_size = kernel_size
+        self.causal = causal
+        self.res_blocks = self.stack_tcn_blocks()
+        self.receptive_fields = self.calculate_receptive_fields()
+        self.res_blocks = nn.Sequential(*self.res_blocks)
+
+    def calculate_receptive_fields(self):
+        receptive_fields = 0
+        for block in self.res_blocks:
+            receptive_fields += block.receptive_fields
+        return receptive_fields
+
+    def build_dilations(self):
+        dilations = []
+        for s in range(0, self.stack_size):
+            for l in range(0, self.stack_num):
+                dilations.append(2**l)
+        return dilations
+
+    def stack_tcn_blocks(self):
+        dilations = self.build_dilations()
+        res_blocks = nn.ModuleList()
+
+        res_blocks.append(
+            TCNBlock(
+                self.in_channels,
+                self.res_channels,
+                self.kernel_size,
+                dilations[0],
+                self.causal,
+            ))
+        for dilation in dilations[1:]:
+            res_blocks.append(
+                TCNBlock(
+                    self.res_channels,
+                    self.res_channels,
+                    self.kernel_size,
+                    dilation,
+                    self.causal,
+                ))
+        return res_blocks
+
+    def forward(self, inputs: torch.Tensor):
+        outputs = inputs
+        outputs = self.res_blocks(outputs)
+        return outputs
+
+
```

Comment:
Any papers about MDTC? please add the  paper titile and link

Reply:
MDTC is new, there is no need to have a paper to ref